src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -917,7 +917,9 @@ private static bool HadLambdaConversionError(DiagnosticBag diagnostics, Analyzed
 
             ParameterSymbol parameter = parm == -1 ? null : method.GetParameters()[parm];
             RefKind refArg = arguments.RefKind(arg);
-            RefKind refParm = parameter?.RefKind ?? RefKind.None; // PROTOTYPE: Ref structs?
+            // PROTOTYPE: Resolve the issue of mixed extension overloads (old and new styles) and RefKinds.
+            var isByRef = receiver?.Type?.IsReferenceType == false && System.Linq.Enumerable.All(symbols, m => m.IsInExtensionClass);
```

Comment:
There were some really wacky things going on with overload resolution here. I forget exactly what they were. (I just tried writing it as an extension method, it somehow gets the fact that `m` is of type char (complains about char not having `IsInExtensionClass`)). I'll try to get a minimal repro eventually and submit an issue.